tests/core/test_lane.py

```diff
@@ -2469,6 +2471,84 @@ def query_works(self, *args, **kwargs):
 
         assert [] == wl.search(self._db, query, RaisesException())
 
+    def test_worklist_for_resultset_no_holds_allowed(self):
```

Comment:
It would be nice to have analogous set of tests for when `DISPLAY_RESERVES` is unset or set to yes/true, but I don't want to hold up this functionality. If we don't get those test cases in this time, we can create a new PR.